Dockerfile

```
@@ -52,16 +52,24 @@ WORKDIR /
 RUN rm -rf cargo-contract
 
 #
-# ink! 4.0 optimizer
-# 
-FROM slimmed-rust as ink-dev
-
-COPY --from=cc-builder /usr/local/bin/cargo-contract /usr/local/bin/cargo-contract
+# ink! types from contract metadata
+#
+FROM slimmed-rust as ink-wrapper
```

I think `ink-dev` is a slightly better name as it's doing lot more than just wrapping:
* it has `cargo contract` capable of building/compiling and testing ink code
* it generates typed API
* etc.

so ink-dev as in _ink development_ is suitable.

---

Overall, `ink-dev` is suitable. Here, I only meant `ink-wrapper` to denote the build stage in which we install `ink-wrapper` (and `gcc` as it will not compile without it).

---

I guess we could pull the `ink-wrapper` part into `ink-dev` and get rid of the separate stage for `ink-wrapper` if this is what you meant.

---

I just meant to leave the old name
```suggestion
FROM slimmed-rust as ink-dev
```